docs/DeveloperGuide.md

```diff
@@ -309,11 +331,339 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 * 3a. The given index is invalid.
 
-    * 3a1. AddressBook shows an error message.
+    * 3a1. ResearchRoster shows an error message.
+
+      Use case resumes at step 2.
+
+
+#### **Use Case: UC03 - Edit a person's record**
+* *Preconditions:* user has added person(s) previously
+* *Guarantees:* person record is edited on the system
+
+**MSS**
+
+1.  User requests a list of persons ([UC05](#use-case-uc05---list-all-persons), [UC06](#use-case-uc06---find-persons-by-criterias)).
+2.  ResearchRoster shows a list of persons.
+3.  User requests to edit a person's record with provided details.
+4.  ResearchRoster edits the person's record.
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. ResearchRoster shows an error message.
+
+      Use case resumes at step 2.
+
+* 3b. The given input is missing all optional fields.
+
+    * 3b1. ResearchRoster shows an error message.
+
+      Use case resumes at step 2.
+
+
+#### **Use Case: UC04 - Edit records in bulk**
+* *Preconditions:* user has added person(s) previously
+* *Guarantees:* person record(s) is/are edited on the system
+
+**MSS**
+
+1.  User requests a filtered list of persons ([UC06](#use-case-uc06---find-persons-by-criterias)).
+2.  ResearchRoster shows the filtered list of persons.
+3.  User requests to apply a change to all records in the list.
+4.  ResearchRoster applies change to all records in the list.
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The change is invalid.
+
+    * 3a1. ResearchRoster shows an error message.
+
+      Use case resumes at step 2.
+
+
+#### **Use Case: UC05 - List all persons**
+* *Preconditions:* user has added person(s) previously
+* *Guarantees:* list of person record(s) is shown
+
+1.  User adds new person(s) to system ([UC01](#use-case-uc01---add-a-new-person)).
```

Comment:
wait how come need step one if the precondition is that user has already added people before?